app/controllers/api/projects.php

```diff
@@ -1600,3 +1604,314 @@
 
         $response->noContent();
     });
+
+// CUSTOM SMTP and Templates
+App::patch('/v1/projects/:projectId/smtp')
+    ->desc('Update SMTP configuration')
+    ->groups(['api', 'projects'])
+    ->label('scope', 'projects.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN])
+    ->label('sdk.namespace', 'projects')
+    ->label('sdk.method', 'updateSmtpConfiguration')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_PROJECT)
+    ->param('projectId', '', new UID(), 'Project unique ID.')
+    ->param('enabled', false, new Boolean(), 'Enable custom SMTP service')
+    ->param('sender', '', new Email(), 'SMTP sender email')
+    ->param('host', '', new HostName(), 'SMTP server host name')
+    ->param('port', null, new Integer(), 'SMTP server port')
+    ->param('username', null, new Text(0), 'SMTP server username')
+    ->param('password', null, new Text(0), 'SMTP server password')
+    ->param('secure', '', new WhiteList(['tls'], true), 'Does SMTP server use secure connection', true)
+    ->inject('response')
+    ->inject('dbForConsole')
+    ->action(function (string $projectId, bool $enabled, string $sender, string $host, int $port, string $username, string $password, string $secure, Response $response, Database $dbForConsole) {
+
+        $project = $dbForConsole->getDocument('projects', $projectId);
+
+        if ($project->isEmpty()) {
+            throw new Exception(Exception::PROJECT_NOT_FOUND);
+        }
+
+        $smtp = [
+            'enabled' => $enabled,
+            'sender' => $sender,
+            'host' => $host,
+            'port' => $port,
+            'username' => $username,
+            'password' => $password,
+            'secure' => $secure,
+        ];
+
+        $project = $dbForConsole->updateDocument('projects', $project->getId(), $project->setAttribute('smtp', $smtp));
+
+        $response->dynamic($project, Response::MODEL_PROJECT);
+    });
+
+App::patch('/v1/projects/:projectId/templates/sms/:type/:locale')
```

Comment:
Let's have the templates GET routes before the PATCH/PUT/DELETE

Reply:
updated.